### DISCUSSION TITLE
Validation markers on renamed files

### DISCUSSION THREAD

**Discussion by cdietrich** [eclipse-langium/langium#1439] [Q&A]
In vscode when you rename a file and that file rename would lead to a validation marker removal, the old validation still stays present. (on the no longer existing file)

i am not sure if this is a vscode problem or a langium problem (send empty diagnostics list for old file?)

https://github.com/eclipse-langium/langium/assets/238736/2d50289a-9b98-4c18-95a8-044e74857fd5
https://github.com/eclipse-langium/langium/assets/238736/d3ced3d5-b594-4bd3-89a0-28d47de9fc2b

https://github.com/cdietrich/langium-rename-problem

**Comment by msujew** [comment]
We don't "unpublish" diagnostics, but we should do that. See https://github.com/eclipse-langium/langium/issues/1440.

On a closer look, it seems like you were already confronted with this issue in the past, https://github.com/microsoft/language-server-protocol/issues/463 :)
